### PR TITLE
quick-webapps: init at 0-unstable-2024-08-05

### PR DESCRIPTION
**Pull request by ahoneybun**
## Description of changes

- homepage URL: https://github.com/elevenhsoft/WebApps#readme-ov-file
- source URL: https://github.com/elevenhsoft/WebApps
- license: gpl3 : https://github.com/elevenhsoft/WebApps#GPL-3.0-1-ov-file
- platforms: linux

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/qu/quick-webapps/package.nix [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      "glyphon-0.5.0" = "sha256-j1HrbEpUBqazWqNfJhpyjWuxYAxkvbXzRKeSouUoPWg=";
+      "smithay-client-toolkit-0.18.0" = "sha256-/7twYMt5/LpzxLXAQKTGNnWcfspUkkZsN5hJu7KaANc";
+      "smithay-clipboard-0.8.0" = "sha256-MqzynFCZvzVg9/Ry/zrbH5R6//erlZV+nmQ2St63Wnc=";
+      "softbuffer-0.4.1" = "sha256-a0bUFz6O8CWRweNt/OxTvflnPYwO5nm6vsyc/WcXyNg=";
+      "taffy-0.3.11" = "sha256-SCx9GEIJjWdoNVyq+RZAGn0N71qraKZxf9ZWhvyzLaI=";
+      "winit-0.29.10" = "sha256-ScTII2AzK3SC8MVeASZ9jhVWsEaGrSQ2BnApTxgfxK4=";
+    };
+  };
+
+  # COSMIC applications now uses vergen for the About page
```
Comment (a-kenji) [inline pkgs/by-name/qu/quick-webapps/package.nix:44]:
```suggestion
  # COSMIC applications now use vergen for the About page
```

**Review by a-kenji — APPROVED** [review]
LGTM, but I guess I have to link to:
#327063

**Review by a-kenji — CHANGES_REQUESTED** [review]
The upstream name changed:
https://github.com/elevenhsoft/WebApps

**Comment by ahoneybun** [comment]
> The upstream name changed: https://github.com/elevenhsoft/WebApps

The developer said that is is WebApps I think?

**Comment by a-kenji** [comment]
Yes, but you are using the name to derive the repo. 
And look at the current repo.

**Comment by ahoneybun** [comment]
> Yes, but you are using the name to derive the repo. And look at the current repo.

Sorry I don't follow?

**Comment by a-kenji** [comment]
> > Yes, but you are using the name to derive the repo. And look at the current repo.
> 
> Sorry I don't follow?

The `repo` argument of the `fetchFromGitHub` function is derived from `pname`, meaning the `src` points to a not anymore existing github repo. Which makes it impossible to build the package.

**Comment by ahoneybun** [comment]
> > > Yes, but you are using the name to derive the repo. And look at the current repo.
> > 
> > 
> > Sorry I don't follow?
> 
> The `repo` argument of the `fetchFromGitHub` function is derived from `pname`, meaning the `src` points to a not anymore existing github repo. Which makes it impossible to build the package.

It built fine for me? I think I'm following but the logic of how it is currently tracks.

**Comment by ahoneybun** [comment]
Oh I see now that is new.

**Review by a-kenji — APPROVED** [review]

File: pkgs/by-name/qu/quick-webapps/package.nix [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  pkg-config,
+  libxkbcommon,
+  openssl,
+  wayland,
+  xorg,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "quick-webapps";
+  version = "master";
```
Comment (SuperSandro2000) [inline pkgs/by-name/qu/quick-webapps/package.nix:18]:
```suggestion
  version = "0.4.6-unstable-2024-06-24";
```

File: pkgs/by-name/qu/quick-webapps/package.nix [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    };
+  };
+
+  # COSMIC applications now use vergen for the About page
+  # Update the COMMIT_DATE to match when the commit was made
+  env.VERGEN_GIT_COMMIT_DATE = "2024-08-03";
+  env.VERGEN_GIT_SHA = src.rev;
+
+  postPatch = ''
+    substituteInPlace justfile --replace '#!/usr/bin/env' "#!$(command -v env)"
```
Comment (SuperSandro2000) [inline pkgs/by-name/qu/quick-webapps/package.nix:51]:
```suggestion
    substituteInPlace justfile --replace-fail '#!/usr/bin/env' "#!$(command -v env)"
```

File: pkgs/by-name/qu/quick-webapps/package.nix [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    "--set"
+    "bin-src"
+    "target/${stdenv.hostPlatform.rust.cargoShortTarget}/release/quick-webapps"
+  ];
+
+  # LD_LIBRARY_PATH can be removed once tiny-xlib is bumped above 0.2.2
+  postInstall = ''
+    wrapProgram "$out/bin/${pname}" \
+      --suffix XDG_DATA_DIRS : "${cosmic-icons}/share" \
+      --prefix LD_LIBRARY_PATH : ${
```
Comment (SuperSandro2000) [inline pkgs/by-name/qu/quick-webapps/package.nix:80]: Can we avoid LD_LIBRARY_PATH by using patchelf?

File: pkgs/by-name/qu/quick-webapps/package.nix [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    "prefix"
+    (placeholder "out")
+    "--set"
+    "bin-src"
+    "target/${stdenv.hostPlatform.rust.cargoShortTarget}/release/quick-webapps"
+  ];
+
+  # LD_LIBRARY_PATH can be removed once tiny-xlib is bumped above 0.2.2
+  postInstall = ''
+    wrapProgram "$out/bin/${pname}" \
```
Comment (SuperSandro2000) [inline pkgs/by-name/qu/quick-webapps/package.nix:78]:
```suggestion
    wrapProgram "$out/bin/quick-webapps" \
```

File: pkgs/by-name/qu/quick-webapps/package.nix [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+          xorg.libXi
+          wayland
+          libxkbcommon
+        ]
+      }
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/elevenhsoft/WebApps";
+    description = "Web applications at your fingertips.";
```
Comment (SuperSandro2000) [inline pkgs/by-name/qu/quick-webapps/package.nix:94]:
```suggestion
    description = "Web applications at your fingertips";
```

**Comment by Thatoo** [comment]
Can it be backported to 24.05 to be integrated into https://github.com/lilyinstarlight/nixos-cosmic flake that indeed work with stable?

File: pkgs/by-name/qu/quick-webapps/package.nix (hunk 4 above)
Comment (ahoneybun) [reply pkgs/by-name/qu/quick-webapps/package.nix:80]: I'm not quite sure about how to go about that but I'm not against if someone does it as long as it works.

Comment (Pandapip1) [reply pkgs/by-name/qu/quick-webapps/package.nix:80]: Add `autoPatchelfHook` to `nativeBuildInputs`.

**Comment by Pandapip1** [comment]
@ahoneybun Did you mean to do this?

**Comment by ahoneybun** [comment]
I don't think I touched this repo so no lol.

**Pull request closed without merging**